### ISSUE TITLE
This function is neat, but why use the ConnectState() decorator at all?

### ISSUE THREAD

**Issue by bbarry** [Rush/ng-connect-state#1]
It seems you could just recommend using `@UntilDestroy()` 

You also don't need to have the interface implemented

```typescript
import { UntilDestroy } from '@ngneat/until-destroy';
import { connectState } from 'ng-connect-state';

@UntilDestroy()
@Component({ template: '{{ state.timer }} <button (click)="state.reload()"</button> Loading: {{ state.loading.timer }}' })
export class InboxComponent {
  /**
   this exposes state.timer as a synchronous value!
  **/
  state = connectState(this, {
    timer: interval(1000),
  })
}
```

(of course you'd have to change the type for the parameter to `connectState`...)

**Comment by Rush** [comment]
Hey @bbarry.  Correct. We could make that recommendation but I also can imagine a situation when we might need to apply more than the `@UntilDestroy()` decorator.

If there was an easy way to have a "ngOnInit" hook for example we would be able to remove this not so sexy try / catch https://github.com/Rush/ng-connect-state/blob/master/src/lib/connect-state.ts#L83

Feel free to just use `@UntilDestroy()` decorator but bear in mind it can change in the future.

**Comment by bbarry** [comment]
Ahh...

I believe the necessary patch set for an ngOnInit hook was pushed to the angular vnext and to 10.0.x last week:

https://github.com/angular/angular/pull/35464

It should be a simple matter of attaching a ngOnInit function to the prototype (which calls an existing one if it exists) in the next point release of Angular.

**Comment by bbarry** [comment]
update: ability to override ngOnInit was added in 10.0.5 yesterday:

https://github.com/angular/angular/blob/master/CHANGELOG.md#1005-2020-07-22

**Comment by Rush** [comment]
Good find @bbarry 
I think I'll wait until until-destroy relies on it.

**Comment by Rush** [comment]
I've just read your comment https://github.com/ngneat/until-destroy/issues/122#issuecomment-680056775 :-(

**Comment by bbarry** [comment]
Yeah, makes me sad too.

From a perf perspective it kinda makes sense that the lifecycle hooks are getting bound during initialization instead of dynamically checked for as they happen but some of them like OnDestroy probably could be late bound without any notable impact. 

I wish there were built in observables to these events that could be hooked into but that probably has perf impacts also.

On the bright side the OnInit hook should work from a class level decorator function >10.0.5 (it just wouldn't work from within a pipe operator or function when the constructor is called I think).

**Comment by Rush** [comment]
> On the bright side the OnInit hook should work from a class level decorator function >10.0.5 (it just wouldn't work from within a pipe operator or function when the constructor is called I think).


 Yeah, I just checked it and I couldn't get rid of the decorator. Oh well....

**Comment by Rush** [comment]
I published 2.0.5 that just relies on `@UntilDestroy()` and supports Angular 13
